src/algebra/algebra/subalgebra.lean

```diff
@@ -288,6 +288,26 @@ we define it as a `linear_equiv` to avoid type equalities. -/
 def to_submodule_equiv (S : subalgebra R A) : S.to_submodule ≃ₗ[R] S :=
 linear_equiv.of_eq _ _ rfl
 
+instance algebra' {P : Type*} [comm_semiring P] [algebra P R] [algebra P A]
```

Comment:
I also suspect you can prove this with weaker assumptions on one of the actions.

Can you prove `algebra` in terms of `algebra'`?